### ISSUE TITLE
Decide what parsing library to use (whole team involvement)

### ISSUE THREAD

**Issue by Mich0608** [uchicago-cs/chiventure#865]
As a team, we need to decide which Python parsing library to use for implementing a domain specific language (DSL). We are currently considering lark, PyParsing, or PLY. We will compare the three libraries and make a decision.

**Comment by Mich0608** [comment]
Our team examined the three libraries lark (#862), PyParsing (#860) and PLY (#859), we met and discussed the pros and cons of each library. By the end of the meeting, the consensus was to use **Lark** as our final parsing library.

These are some of the things we liked about Lark:

1. Lark performs automated tokenization
2. It is well-documented
3. It has a relatively intuitive and straightforward syntax
4. It supports ambiguous grammar and can ignore white spaces which could be potentially beneficial during our implementation
5. It is a popular parsing library, so it will be relatively easy to ask questions about it online if necessary

These are the major reasons we decided **NOT** to use PLY and PyParsing:

- We decided NOT to use PLY because PLY requires a two-step process of first tokenizing the input string then parsing it into the desired outcome, whereas the other two libraries automatically tokenize the input.

- We decided NOT to use PyParsing because the syntax wasn't as intuitive compared to Lark.
